src/plugins/intel_gpu/src/kernel_selector/cl_kernels/slice_ref.cl

```diff
@@ -4,33 +4,89 @@
 
 #include "include/batch_headers/fetch_data.cl"
 
-KERNEL(slice_ref)(const __global INPUT0_TYPE* input, __global OUTPUT_TYPE* output)
+#define BRING_INTO_RANGE(VAL, MAX) \
+    clamp((long)VAL < 0l ? (long)VAL + (long)MAX : (long)VAL, 0l, (long)MAX-1l);
+
+#if INPUT0_DIMS < 5
+#define LOAD_BUFFER(in_prefix, out_name)  \
+    long out_name[INPUT0_DIMS];           \
+    out_name[0] = in_prefix##_VAL0;       \
+    out_name[1] = in_prefix##_VAL1;       \
+    out_name[2] = in_prefix##_VAL2;       \
+    out_name[3] = in_prefix##_VAL3;
+#else
+#define LOAD_BUFFER(in_prefix, out_name)  \
+    long out_name[INPUT0_DIMS];           \
+    out_name[0] = in_prefix##_VAL0;       \
+    out_name[1] = in_prefix##_VAL1;       \
+    out_name[2] = in_prefix##_VAL2;       \
+    out_name[3] = in_prefix##_VAL3;       \
+    out_name[4] = in_prefix##_VAL4;
+#endif
+
+KERNEL(slice_ref)(OPTIONAL_SHAPE_INFO_ARG 
+                  const __global INPUT0_TYPE* restrict input,
+                  START_BUFFER
+                  STEP_BUFFER
+                  AXES_BUFFER
+                  __global OUTPUT_TYPE* restrict output)
 {
-    const uint batch = get_global_id(0);
-    const uint feature = get_global_id(1);
+    LOAD_BUFFER(START, start_buff);
+    LOAD_BUFFER(STEP, step_buff);
+    LOAD_BUFFER(AXES, axes_buff);
+
+    long slice_step[INPUT0_DIMS];
+    long slice_start[INPUT0_DIMS];
+    #pragma unroll
+    for(int i = 0; i < INPUT0_DIMS; ++i) {
+        slice_step[i] = 1;
+        slice_start[i] = 0;
+    }
+
+    #pragma unroll
```

Comment:
nit: usually we use GPU Plugin's macro `unroll_for` for unrolling hint

Reply:
OK, fixed.